src/std/iter.ss

```diff
@@ -129,17 +129,35 @@ package: std
             (set! (cdr e) (##fx- limit 1))
             value)
           iter-end))))
-  (if (and (number? start) (integer? count) (number? step))
+  (if (and (number? start) (fixnum? count) (number? step))
     (make-iterator (cons start count) next)
-    (error "Parameters are of wrong type (count:number start:integer step:number)."
+    (error "Parameters are of wrong type (count:fixnum start:number step:number)."
       [count start step])))
 
-(def* in-range
+(def* in-iota
   ((count) (iter-in-range 0 count 1))
   ((count start) (iter-in-range start count 1))
   ((count start step) (iter-in-range start count step)))
 
-(def (in-naturals (start 1) (step 1))
+(def (iter-in-range start end step)
```

Comment:
Here you change the style from the `iter-in-iota` a little. Could you do:

```scheme
(def (iter-in-range start end step)
  (def (next it)
    (with ((iterator e) it)
      (with ([value . limit] e)
        (let ((value+step (+ value step)))
          (if (##fx<= value+step limit)
            (begin
              (set! (car e) value+step)
              value)
            iter-end)))))
  (if (and (number? start) (integer? end) (number? step))
    (make-iterator (cons start end) next)
    (error "Parameters are of wrong type (count:number start:integer step:number)."
      [start end step])))
```
?

Reply:
The `begin` can be put on one line as well, I find it more readable as it is short.

Reply:
The difference is whether the next function is a closure (in the new code) or a liftable procedure that uses the iterator as the closure.  Semantically they are the same.

Reply:
Note that having a liftable procedure is slightly more efficient (in the in-iota code).

Reply:
Nonetheless, I think fare's way is more natural and easier to read (also it's simpler).
@fare may I suggest you modify the old code to be written in the same style?
Let's be consistent.

Reply:
Fine with me as long as it's consistent.

Reply:
actually we need to keep both the count and the value in iota, so we can't write it the same way with a closure; nm.